pkg/sources/filesystem/filesystem.go

```diff
@@ -106,13 +108,37 @@ func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk) err
 			}
 
 			inputFile, err := os.Open(path)
+			defer inputFile.Close()
 			if err != nil {
 				log.Warn(err)
 				return nil
 			}
 			defer inputFile.Close()
 
+			chunkSkel := &sources.Chunk{
+				SourceType: s.Type(),
+				SourceName: s.name,
+				SourceID:   s.SourceID(),
+				SourceMetadata: &source_metadatapb.MetaData{
+					Data: &source_metadatapb.MetaData_Filesystem{
+						Filesystem: &source_metadatapb.Filesystem{
+							File: sanitizer.UTF8(path),
+						},
+					},
+				},
+				Verify: s.verify,
+			}
+			if handlers.HandleFile(inputFile, chunkSkel, chunksChan) {
+				return nil
+			}
+
+			if err != nil && !errors.Is(err, archiver.ErrNoMatch) {
+				log.WithError(err).Debug("Error reading archive")
+				return nil
+			}
+
 			reader := bufio.NewReaderSize(bufio.NewReader(inputFile), BufferSize)
```

Comment:
At this point it's only safe to continue here if the reader is consumed (cursor is at EOF, processed by the handler) or if the original reader is untouched (no cursor advance)

Reply:
Ended up with a mix of a couple of methods here. Got it all sorted out.